### PR TITLE
feat(gRPC): Make a single gRPC package and improve interceptor log messages

### PR DESCRIPTION
**Pull request by EduardGomezEscandell**
1. clientID is now customizable.
2. coalesce the two gRPC packages into one to prevent them from diverging.
3. print the clientID when the connection closes.
4. Use (1) to set the clientID as the distroname.

---

This PR is a spin-off from #563

### CODE REVIEW

**Comment by EduardGomezEscandell** [comment]
Not sure how to cover the WithoutRemoteSend function

**Review by CarlosNihelton — APPROVED** [review]
LGTM ocen the typo is fixed

File: wsl-pro-service/internal/system/system_test.go [hunk 1]
```diff
@@ -120,6 +105,59 @@ func TestInfo(t *testing.T) {
...
+			}
+
+			got, err := system.WslDistroName(ctx)
+			if tc.wantErr {
+				require.Error(t, err, "Expected Get() to return an error")
+				return
+			}
+			require.NoError(t, err, "Expected Get() to return no errors")
+
+			assert.Equal(t, "TEST_DISTRO", got, "WslName does not match expected value")
```
[the remark this answers is not in the document]
Comment (EduardGomezEscandell) [reply wsl-pro-service/internal/system/system_test.go:155]: Yeah, good catch 🔍

File: wsl-pro-service/internal/controlstream/controlstream_test.go [hunk 2]
```diff
@@ -84,14 +84,16 @@ func TestConnect(t *testing.T) {
 	testCases := map[string]struct {
 		portFile              dataFileState
 		breakWindowsLocalhost bool
+		breakWSlDisroName     bool
```
[the remark this answers is not in the document]
Comment (EduardGomezEscandell) [reply wsl-pro-service/internal/controlstream/controlstream_test.go:87]: I cannot be trusted with a keyboard 😭😄

[the remark this answers is not in the document]
Comment (CarlosNihelton) [reply wsl-pro-service/internal/controlstream/controlstream_test.go:87]: I wish I could spot my own typos as I can spot others....

**Comment by EduardGomezEscandell** [comment]
Rebased to fix merge conflicts. New changes start now.

**Pull request merged**